pkgs/applications/file-managers/yazi/default.nix

```diff
@@ -0,0 +1,102 @@
+{ rustPlatform
+, fetchFromGitHub
+, lib
+
+, withFile ? true
+, file
+, withJq ? true
+, jq
+, withPoppler ? true
+, poppler_utils
+, withUnar ? true
+, unar
+, withFfmpegthumbnailer ? true
+, ffmpegthumbnailer
+, withFd ? true
+, fd
+, withRipgrep ? true
+, ripgrep
+, withFzf ? true
+, fzf
+, withZoxide ? true
+, zoxide
+
+, stdenv
+, Foundation
+
+, nix-update-script
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "yazi";
+  version = "0.1.3";
+
+  src = fetchFromGitHub {
+    owner = "sxyazi";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-IUE2846AltYyEefkavCJOEak9mW0wygojWsucwUEgh4=";
+  };
+
+  postPatch =
```

Comment:
I see what your intention was here, but that's not the way it should be done IMO!

I think what you should do is something like this:

```nix
installPhase = let
  runtimePaths = []
  ++ (lib.optionals withFile [ file ])
  # ... others accordingly
  ;
in
''
    runHook preInstall

    wrapProgram $out/bin/yazi \
        --prefix PATH : "${lib.makeBinPath runtimePaths}"

    runHook postInstall
'';

(above is untested).

Can you check whether this works as expected and update your PR?